src/merkle/merkle_tree.rs

```diff
@@ -200,49 +200,53 @@ mod tests {
     }
 
     #[test]
-    fn get_leaf() {
+    fn get_leaf() -> Result<(), MerkleError> {
         let tree = super::MerkleTree::new(LEAVES4.to_vec()).unwrap();
 
         // check depth 2
-        assert_eq!(LEAVES4[0], tree.get_node(NodeIndex::new(2, 0)).unwrap());
-        assert_eq!(LEAVES4[1], tree.get_node(NodeIndex::new(2, 1)).unwrap());
-        assert_eq!(LEAVES4[2], tree.get_node(NodeIndex::new(2, 2)).unwrap());
-        assert_eq!(LEAVES4[3], tree.get_node(NodeIndex::new(2, 3)).unwrap());
+        assert_eq!(LEAVES4[0], tree.get_node(NodeIndex::new(2, 0)?).unwrap());
```

Comment:
I'm not sure it makes sense to change the function to return a result and still `unwrap`. We should either go for full `?` or `unwrap/expect`.